Public commit 1b97e2dfd1bf3bac2d67bcbfc91de14412c9a6d6 -> c59d7985c4c19204b8d4bcda2c4e6cfd4082a5bf
--- 1b97e2dfd1bf3bac2d67bcbfc91de14412c9a6d6:token-metadata/program/src/processor/uses/utilize.rs
+++ c59d7985c4c19204b8d4bcda2c4e6cfd4082a5bf:token-metadata/program/src/processor/uses/utilize.rs
@@ -54,7 +54,7 @@
     if metadata.uses.is_none() {
         return Err(MetadataError::Unusable.into());
     }
-    if *token_program_account_info.key != spl_token::id() {
+    if *token_program_account_info.key != spl_token::ID {
         return Err(MetadataError::InvalidTokenProgram.into());
     }
     assert_signer(user_info)?;

Public commit c59d7985c4c19204b8d4bcda2c4e6cfd4082a5bf -> 3a4b3f6b51c956c7bab7aeae124faff43513dfaa
--- c59d7985c4c19204b8d4bcda2c4e6cfd4082a5bf:token-metadata/program/src/processor/uses/utilize.rs
+++ 3a4b3f6b51c956c7bab7aeae124faff43513dfaa:token-metadata/program/src/processor/uses/utilize.rs
@@ -30,6 +30,7 @@
     number_of_uses: u64,
 ) -> ProgramResult {
     let account_info_iter = &mut accounts.iter().peekable();
+
     let metadata_info = next_account_info(account_info_iter)?;
     let token_account_info = next_account_info(account_info_iter)?;
     let mint_info = next_account_info(account_info_iter)?;
@@ -37,7 +38,8 @@
     let owner_info = next_account_info(account_info_iter)?;
     let token_program_account_info = next_account_info(account_info_iter)?;
     let _ata_program_account_info = next_account_info(account_info_iter)?;
-    let _system_account_info = next_account_info(account_info_iter)?;
+    let _system_program_account_info = next_account_info(account_info_iter)?;
+
     // consume the next account only if it is Rent
     let approved_authority_is_using = if account_info_iter
         .next_if(|info| info.key == &Rent::id())
@@ -51,6 +53,7 @@
     };
 
     let metadata: Metadata = Metadata::from_account_info(metadata_info)?;
+
     if metadata.uses.is_none() {
         return Err(MetadataError::Unusable.into());
     }

Public commit 3a4b3f6b51c956c7bab7aeae124faff43513dfaa -> 887708157d86addf0cfd87c15de19ef1eb984a87
--- 3a4b3f6b51c956c7bab7aeae124faff43513dfaa:token-metadata/program/src/processor/uses/utilize.rs
+++ 887708157d86addf0cfd87c15de19ef1eb984a87:programs/token-metadata/program/src/processor/uses/utilize.rs
@@ -109,7 +109,7 @@
     } else if user_info.key != owner_info.key {
         return Err(MetadataError::InvalidUser.into());
     }
-    metadata.serialize(&mut *metadata_info.try_borrow_mut_data()?)?;
+    metadata.save(&mut metadata_info.try_borrow_mut_data()?)?;
     if remaining_uses == 0 && must_burn {
         if approved_authority_is_using {
             let burn_authority_info = next_account_info(account_info_iter)?;

Public commit 887708157d86addf0cfd87c15de19ef1eb984a87 -> 40421f05e99de309690a04e737e607f956560fca
--- 887708157d86addf0cfd87c15de19ef1eb984a87:programs/token-metadata/program/src/processor/uses/utilize.rs
+++ 40421f05e99de309690a04e737e607f956560fca:programs/token-metadata/program/src/processor/uses/utilize.rs
@@ -22,6 +22,7 @@
     },
     error::MetadataError,
     state::{Metadata, TokenMetadataAccount, UseAuthorityRecord, UseMethod, Uses, BURN, PREFIX},
+    utils::SPL_TOKEN_ID,
 };
 
 pub fn process_utilize(
@@ -57,7 +58,7 @@
     if metadata.uses.is_none() {
         return Err(MetadataError::Unusable.into());
     }
-    if *token_program_account_info.key != spl_token::ID {
+    if *token_program_account_info.key != SPL_TOKEN_ID {
         return Err(MetadataError::InvalidTokenProgram.into());
     }
     assert_signer(user_info)?;

Public commit 40421f05e99de309690a04e737e607f956560fca -> 2110c34f83a997576507ccf4e810a0a29e69a8c2
--- 40421f05e99de309690a04e737e607f956560fca:programs/token-metadata/program/src/processor/uses/utilize.rs
+++ 2110c34f83a997576507ccf4e810a0a29e69a8c2:programs/token-metadata/program/src/processor/uses/utilize.rs
@@ -1,4 +1,3 @@
-use borsh::BorshSerialize;
 use mpl_utils::{
     assert_signer,
     token::{spl_token_burn, TokenBurnParams},
@@ -22,7 +21,6 @@
     },
     error::MetadataError,
     state::{Metadata, TokenMetadataAccount, UseAuthorityRecord, UseMethod, Uses, BURN, PREFIX},
-    utils::SPL_TOKEN_ID,
 };
 
 pub fn process_utilize(
@@ -58,7 +56,7 @@
     if metadata.uses.is_none() {
         return Err(MetadataError::Unusable.into());
     }
-    if *token_program_account_info.key != SPL_TOKEN_ID {
+    if *token_program_account_info.key != spl_token::ID {
         return Err(MetadataError::InvalidTokenProgram.into());
     }
     assert_signer(user_info)?;
@@ -106,7 +104,7 @@
             .allowed_uses
             .checked_sub(number_of_uses)
             .ok_or(MetadataError::NotEnoughUses)?;
-        record.serialize(data)?;
+        borsh::to_writer(&mut data[..], &record)?;
     } else if user_info.key != owner_info.key {
         return Err(MetadataError::InvalidUser.into());
     }

Public commit 2110c34f83a997576507ccf4e810a0a29e69a8c2 -> 4e5bcca44000f151fe64682826bbe2eb61cd7b87
--- 2110c34f83a997576507ccf4e810a0a29e69a8c2:programs/token-metadata/program/src/processor/uses/utilize.rs
+++ 4e5bcca44000f151fe64682826bbe2eb61cd7b87:programs/token-metadata/program/src/processor/uses/utilize.rs
@@ -21,6 +21,7 @@
     },
     error::MetadataError,
     state::{Metadata, TokenMetadataAccount, UseAuthorityRecord, UseMethod, Uses, BURN, PREFIX},
+    utils::SPL_TOKEN_ID,
 };
 
 pub fn process_utilize(
@@ -56,7 +57,7 @@
     if metadata.uses.is_none() {
         return Err(MetadataError::Unusable.into());
     }
-    if *token_program_account_info.key != spl_token::ID {
+    if *token_program_account_info.key != SPL_TOKEN_ID {
         return Err(MetadataError::InvalidTokenProgram.into());
     }
     assert_signer(user_info)?;

Public commit 4e5bcca44000f151fe64682826bbe2eb61cd7b87 -> 6d1d6ff4669cbadb86f42c2606f01269a8b26495
--- 4e5bcca44000f151fe64682826bbe2eb61cd7b87:programs/token-metadata/program/src/processor/uses/utilize.rs
+++ 6d1d6ff4669cbadb86f42c2606f01269a8b26495:programs/token-metadata/program/src/processor/uses/utilize.rs
@@ -2,8 +2,8 @@
     assert_signer,
     token::{spl_token_burn, TokenBurnParams},
 };
-use solana_program::{
-    account_info::{next_account_info, AccountInfo},
+use arch_program::{
+    account::{next_account_info, AccountInfo},
     entrypoint::ProgramResult,
     pubkey::Pubkey,
     rent::Rent,
@@ -29,19 +29,19 @@
     accounts: &[AccountInfo],
     number_of_uses: u64,
 ) -> ProgramResult {
-    let account_info_iter = &mut accounts.iter().peekable();
+    let account_iter = &mut accounts.iter().peekable();
 
-    let metadata_info = next_account_info(account_info_iter)?;
-    let token_account_info = next_account_info(account_info_iter)?;
-    let mint_info = next_account_info(account_info_iter)?;
-    let user_info = next_account_info(account_info_iter)?;
-    let owner_info = next_account_info(account_info_iter)?;
-    let token_program_account_info = next_account_info(account_info_iter)?;
-    let _ata_program_account_info = next_account_info(account_info_iter)?;
-    let _system_program_account_info = next_account_info(account_info_iter)?;
+    let metadata_info = next_account_info(account_iter)?;
+    let token_account = next_account_info(account_iter)?;
+    let mint_info = next_account_info(account_iter)?;
+    let user_info = next_account_info(account_iter)?;
+    let owner_info = next_account_info(account_iter)?;
+    let token_program_account = next_account_info(account_iter)?;
+    let _ata_program_account = next_account_info(account_iter)?;
+    let _system_program_account = next_account_info(account_iter)?;
 
     // consume the next account only if it is Rent
-    let approved_authority_is_using = if account_info_iter
+    let approved_authority_is_using = if account_iter
         .next_if(|info| info.key == &Rent::id())
         .is_some()
     {
@@ -52,12 +52,12 @@
         accounts.len() == 10
     };
 
-    let metadata: Metadata = Metadata::from_account_info(metadata_info)?;
+    let metadata: Metadata = Metadata::from_account(metadata_info)?;
 
     if metadata.uses.is_none() {
         return Err(MetadataError::Unusable.into());
     }
-    if *token_program_account_info.key != SPL_TOKEN_ID {
+    if *token_program_account.key != SPL_TOKEN_ID {
         return Err(MetadataError::InvalidTokenProgram.into());
     }
     assert_signer(user_info)?;
@@ -67,9 +67,9 @@
         metadata_info,
         &metadata,
         mint_info,
-        token_account_info,
+        token_account,
     )?;
-    let mut metadata = Metadata::from_account_info(metadata_info)?;
+    let mut metadata = Metadata::from_account(metadata_info)?;
     let metadata_uses = metadata.uses.unwrap();
     let must_burn = metadata_uses.use_method == UseMethod::Burn;
     if number_of_uses > metadata_uses.total || number_of_uses > metadata_uses.remaining {
@@ -85,7 +85,7 @@
         remaining: remaining_uses,
     });
     if approved_authority_is_using {
-        let use_authority_record_info = next_account_info(account_info_iter)?;
+        let use_authority_record_info = next_account_info(account_iter)?;
         let data = &mut *use_authority_record_info.try_borrow_mut_data()?;
         process_use_authority_validation(data.len(), false)?;
         assert_owned_by(use_authority_record_info, program_id)?;
@@ -112,7 +112,7 @@
     metadata.save(&mut metadata_info.try_borrow_mut_data()?)?;
     if remaining_uses == 0 && must_burn {
         if approved_authority_is_using {
-            let burn_authority_info = next_account_info(account_info_iter)?;
+            let burn_authority_info = next_account_info(account_iter)?;
             let seed = assert_burner(burn_authority_info.key)?;
             let burn_bump_ref = &[
                 PREFIX.as_bytes(),
@@ -124,8 +124,8 @@
                 mint: mint_info.clone(),
                 amount: 1,
                 authority: burn_authority_info.clone(),
-                token_program: token_program_account_info.clone(),
-                source: token_account_info.clone(),
+                token_program: token_program_account.clone(),
+                source: token_account.clone(),
                 authority_signer_seeds: Some(burn_bump_ref),
             })?;
         } else {
@@ -133,8 +133,8 @@
                 mint: mint_info.clone(),
                 amount: 1,
                 authority: owner_info.clone(),
-                token_program: token_program_account_info.clone(),
-                source: token_account_info.clone(),
+                token_program: token_program_account.clone(),
+                source: token_account.clone(),
                 authority_signer_seeds: None,
             })?;
         }
